controllers/configmap_controller.go

```diff
@@ -387,9 +393,21 @@ func (r *ConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			builder.WithPredicates(outdatedWindowsNodePredicate(true))).
 		Watches(&source.Kind{Type: &core.Node{}}, handler.EnqueueRequestsFromMapFunc(r.mapToServicesConfigMap),
 			builder.WithPredicates(windowsNodeVersionChangePredicate())).
+		// An update to the rendered worker MachineConfig could include changes to the kubelet args, which must be
+		// reflected in the services ConfigMap.
+		Watches(&source.Kind{Type: &mcfg.MachineConfig{}}, handler.EnqueueRequestsFromMapFunc(r.mapToServicesConfigMap),
```

Comment:
Yes, that is me thinking too far ahead. I've changed the predicate to filter out all MachineConfig events, leaving it in the `Watches` is important as it allows the cached client to pull the MachineConfig object. Added a [TODO](https://issues.redhat.com/browse/WINC-897) to stop filtering events and act upon changes.

Reply:
> leaving it in the Watches is important as it allows the cached client to pull the MachineConfig object

That sounds really weird. This shouldn't be the reason to Watch the object. 

>Added a [TODO](https://issues.redhat.com/browse/WINC-897) to stop filtering events and act upon changes.

This sounds like a risky TODO. Do we know how often these rendered config changes? We don't want a situation where we are unnecessarily re-configuring Windows nodes.

Reply:
+1 on removing this watch and `renderedWorkerPredicate`. You can put the TODO with issue link here, and if you'd like update the issue engineering details to capture the watch + predicate.

Reply:
I continue to find this strange. It is one thing to say we need watch for changes in the future but another to add it here because it allows the cached client to pull the MachineConfig object without knowing why. Are you sure you are not missing adding it to a scheme somewhere in the code?